x-pack/plugins/alerting/server/data/ad_hoc_run/types/ad_hoc_run.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RuleDomain } from '../../../application/rule/types';
+import { AdHocRunStatus } from '../../../../common/constants';
+
+export interface AdHocRunSchedule extends Record<string, unknown> {
+  interval: string;
+  status: AdHocRunStatus;
+  runAt: string;
+}
+
+// This is the rule information stored in the AD_HOC_RUN_SAVED_OBJECT_TYPE saved object
```

Comment:
That's interesting! So currently if a user schedules a backfill for rule `A` for Jan 1 00:00 - 05:00, there might be 5 executions in the schedule and they would get a backfill ID of `1` in the response that I planned to use as the scheduled task ID (in the followup PR).

Then later if they schedule a backfill for rule `A` for Jan 2 10:00 - 12:00, there would be 2 executions in the schedule. In the current PR implementation, they would get a different backfill ID for this request and a different task would be scheduled to handle those 2 execution sets.

Is your suggestion that instead of generating a different ID and task for the second request, we would just try to update the schedule for backfill ID `1` to include the 2 new executions? 

I am thinking the current approach would align better with the planned gap management features as you could correlate a specific gap with a specific backfill ID instead of parsing through the schedule for a time range but this is something that could possibly evolve as I work through the task runner portion of the feature :)